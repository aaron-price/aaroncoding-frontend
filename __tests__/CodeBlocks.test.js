
/* eslint-env jest */

const { shallow } = require('enzyme')
const React = require('react')
const renderer = require('react-test-renderer')
const App = require('../components/CodeBlocks.js').CodeBlock
import Adapter from 'enzyme-adapter-react-16'
import { configure } from 'enzyme'
configure({ adapter: new Adapter() })

describe('With Enzyme', () => {
    it('says hi', () => {
        const app = shallow(<App><p>hi</p></App>)
        expect(app.find('p').text()).toEqual('hi')
    })
    it('has the correct class', () => {
        const app = shallow(<App><p>hi</p></App>)
        expect(app.find('.codeblock').exists()).toBe(true)
    })
})

describe('With Snapshot Testing', () => {
    it('App shows "hi"', () => {
        const component = renderer.create(<App />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})