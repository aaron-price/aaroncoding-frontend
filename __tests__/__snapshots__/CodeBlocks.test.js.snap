// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "hi" 1`] = `
<pre
  className="codeblock"
>
  <code />
</pre>
`;
